lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,336 @@
+  await collectionsModel.createTable();
+
+  providersModel = new Provider();
+  await providersModel.createTable();
+
+  pdrsModel = new Pdr();
+  await pdrsModel.createTable();
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+});
+
+test.beforeEach(async (t) => {
+  const testCollection = fakeCollectionFactory();
+  const testProvider = fakeProviderFactory({
+    encrypted: true,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+  });
+
+  t.context.testCollection = testCollection;
+  t.context.testProvider = testProvider;
+
+  t.context.collectionCumulusId = await migrateFakeCollectionRecord(testCollection, t.context.knex);
+  t.context.providerCumulusId = await migrateFakeProviderRecord(testProvider, t.context.knex);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex(tableNames.pdrs).del();
+  await t.context.knex(tableNames.providers).del();
+  await t.context.knex(tableNames.collections).del();
+});
+
+test.after.always(async (t) => {
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await pdrsModel.deleteTable();
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migratePdrRecord correctly migrates PDR record', async (t) => {
+  const { knex, testCollection, testProvider, collectionCumulusId, providerCumulusId } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.id);
+  await migratePdrRecord(testPdr, knex);
+
+  const record = await t.context.knex.queryBuilder()
+    .select()
+    .table(tableNames.pdrs)
+    .where({ name: testPdr.pdrName })
+    .first();
+
+  t.like(
+    omit(record, ['cumulus_id']),
+    omit({
+      name: testPdr.pdrName,
+      provider_cumulus_id: providerCumulusId,
+      collection_cumulus_id: collectionCumulusId,
+      execution_cumulus_id: testPdr.executionCumulusId ? testPdr.executionCumulusId : null,
+      status: testPdr.status,
+      progress: testPdr.progress,
+      pan_sent: testPdr.PANSent,
+      pan_message: testPdr.PANmessage,
+      stats: testPdr.stats,
+      address: testPdr.address,
+      original_url: testPdr.originalUrl,
+      timestamp: new Date(testPdr.timestamp),
+      duration: testPdr.duration,
+      created_at: new Date(testPdr.createdAt),
+      updated_at: new Date(testPdr.updatedAt),
+    },
+    ['updated_at'])
```

Comment:
Do we need to omit `updated_at`? From the code in the test it seems like we know the expected value?

Reply:
Ah, I need to omit it because the time is slightly off, ie:

```
  -   updated_at: Date 2021-01-22 04:09:04 957ms UTC {},
  +   updated_at: Date 2021-01-22 04:09:04 907ms UTC {},
```

but I will remove `updated_at` from the comparison anyway while leaving the omit.

Reply:
Hmm, really? I would expect this adding this to the expected object should work:

`updated_at: new Date(testPdr.updatedAt)`

Reply:
Yeah, I added that and got :
```
  -   updated_at: Date 2021-01-22 20:50:42 618ms UTC {},
  +   updated_at: Date 2021-01-22 20:50:42 585ms UTC {},
```